tests/project/test_mbed_program.py

```diff
@@ -10,17 +10,20 @@
 from mbed_tools.project import MbedProgram
 from mbed_tools.project.exceptions import ExistingProgram, ProgramNotFound, MbedOSNotFound
 from mbed_tools.project.mbed_program import _find_program_root, parse_url
-from mbed_tools.project._internal.project_data import MbedProgramFiles
+from mbed_tools.project._internal.project_data import CMAKE_CONFIG_FILE_NAME, MbedProgramFiles
 from tests.project.factories import make_mbed_program_files, make_mbed_os_files, patchfs
 
 
 DEFAULT_BUILD_SUBDIR = pathlib.Path("K64F", "develop", "GCC_ARM")
+DEFAULT_CONFIG_SUBDIR = pathlib.Path("K64F", "config", "GCC_ARM")
```

Comment:
Shouldn't the default config subdir be `.mbedbuild` or `.`? Which path relative to the project does CMake read mbed_config.cmake from usually?

Reply:
CMake does use the config from `.mbedbuild`, while this would be the path to the retained config for the current target and toolchain. It would be clearer if I were to revert cmake_config_file to `.mbedbuild/mbed_config.cmake` and then just make the backup during `configure`, as the retained config isn't used.

Reply:
Why do we need to "retain the config for the current target and toolchain" if we make the user always pass target and toolchain? This backup stuff sounds unnecessary to me.

Reply:
Just to allow the user to check the previously generated config that is associated with each build, without needing to generate each one again, if that isn't necessary I can drop it.